sympy/printing/pycode.py

```diff
@@ -497,7 +497,7 @@ def _print_seq(self, seq):
 
     def _print_MatMul(self, expr):
         "Matrix multiplication printer"
-        return '({0})'.format(').dot('.join(self._print(i) for i in expr.args))
+        return '({0})'.format(').dot('.join([self._print(expr.args[1]), self._print(expr.args[0])]))
```

Comment:
The first argument of MatMul is only a constant if it is not 1. Otherwise, they are all matrices, and this will be incorrect as matrix multiplication is not commutative. 

It would be better to pull out the non-matrix factor if it exists and use `*` to apply it.

Reply:
But for numpy printer MatMul is defined as a dot product of matrices and dot product is commutative so I think it is correct.